questionnaire_state.tf

```diff
@@ -0,0 +1,178 @@
+  max_capacity       = "${var.questionnaire_state_max_read_capacity}"
+  min_capacity       = "${var.questionnaire_state_min_read_capacity}"
+  resource_id        = "table/${aws_dynamodb_table.questionnaire_state_table.name}"
+  scalable_dimension = "dynamodb:table:ReadCapacityUnits"
+  service_namespace  = "dynamodb"
+}
+
+resource "aws_appautoscaling_policy" "questionnaire_state_table_read_policy" {
+  name               = "DynamoDBReadCapacityUtilization:${aws_appautoscaling_target.questionnaire_state_table_read_target.resource_id}"
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = "${aws_appautoscaling_target.questionnaire_state_table_read_target.resource_id}"
+  scalable_dimension = "${aws_appautoscaling_target.questionnaire_state_table_read_target.scalable_dimension}"
+  service_namespace  = "${aws_appautoscaling_target.questionnaire_state_table_read_target.service_namespace}"
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "DynamoDBReadCapacityUtilization"
+    }
+
+    target_value = 40
+  }
+}
+
+resource "aws_appautoscaling_target" "questionnaire_state_table_write_target" {
+  max_capacity       = "${var.questionnaire_state_max_write_capacity}"
+  min_capacity       = "${var.questionnaire_state_min_write_capacity}"
+  resource_id        = "table/${aws_dynamodb_table.questionnaire_state_table.name}"
+  scalable_dimension = "dynamodb:table:WriteCapacityUnits"
+  service_namespace  = "dynamodb"
+}
+
+resource "aws_appautoscaling_policy" "questionnaire_state_table_write_policy" {
+  name               = "DynamoDBReadCapacityUtilization:${aws_appautoscaling_target.questionnaire_state_table_write_target.resource_id}"
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = "${aws_appautoscaling_target.questionnaire_state_table_write_target.resource_id}"
+  scalable_dimension = "${aws_appautoscaling_target.questionnaire_state_table_write_target.scalable_dimension}"
+  service_namespace  = "${aws_appautoscaling_target.questionnaire_state_table_write_target.service_namespace}"
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "DynamoDBWriteCapacityUtilization"
+    }
+
+    target_value = 40
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "questionnaire_state_table_read_throughput" {
+  alarm_name          = "${var.env}-dynamodb-questionnaire-state-read-throughput"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "1"
+  metric_name         = "ConsumedReadCapacityUnits"
+  namespace           = "AWS/DynamoDB"
+  period              = "60"
+  statistic           = "Sum"
+  threshold           = "${var.questionnaire_state_max_read_capacity * 0.8 * 60}"
+  alarm_description   = "Questionnaire State DynamoDB average consumed read throughput above 80% of maximum for past 60 seconds"
```

Comment:
You should prefix these alarms with EQ as per the session alarms. as in preprod and prod they go to an SDC wide ops channel